### DISCUSSION TITLE
Setting up different compilerPath for C and Cpp

### DISCUSSION THREAD

**Discussion by wytsai7660** [microsoft/vscode-cpptools#11848] [Q&A] [answered]
Hello, I'm struggling with setting up different compiler paths for C and C++.
I'm using Mac, and the paths for `gcc` and `g++` are `"/opt/homebrew/bin/gcc-13"` and `"/opt/homebrew/bin/g++-13"`, respectively. But when I tried to specify the paths like this:
```json
"compilerPath": ["/opt/homebrew/bin/gcc-13", "/opt/homebrew/bin/g++-13"]
```
I got an error: `Incorrect type. Expected "string".`

So what's the correct way to handle this?

**Comment by bobbrow** [comment, marked as answer]
With the right arguments, gcc can forward compilation to g++.  Our extension can take care of this part for IntelliSense, so setting compilerPath to `"/opt/homebrew/bin/gcc-13"` should work.  If not, please let us know as that would be a bug.

**Reply by wytsai7660** [reply]
I see. Thank you very much!
